packages/venia-ui/lib/components/CartPage/ProductListing/EditModal/productForm.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import gql from 'graphql-tag';
+import { Form } from 'informed';
+import { useProductForm } from '@magento/peregrine/lib/talons/CartPage/ProductListing/EditModal/useProductForm';
+
+import { mergeClasses } from '../../../../classify';
+import Button from '../../../Button';
+import LoadingIndicator from '../../../LoadingIndicator';
+import Options from '../../../ProductOptions';
+import { QuantityFields } from '../quantity';
+import defaultClasses from './productForm.css';
+import { CartPageFragment } from '../../cartPageFragments';
+
+const ProductForm = props => {
+    const { item: cartItem, setIsUpdating } = props;
+    const talonProps = useProductForm({
+        cartItem,
+        getConfigurableOptionsQuery: GET_CONFIGURABLE_OPTIONS,
+        setIsUpdating,
+        updateConfigurableOptionsMutation: UPDATE_CONFIGURABLE_OPTIONS_MUTATION,
+        updateQuantityMutation: UPDATE_QUANTITY_MUTATION
+    });
+    const {
+        configItem,
+        handleOptionSelection,
+        handleSubmit,
+        isLoading,
+        setFormApi
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    if (isLoading) {
+        return (
+            <LoadingIndicator
+                classes={{ root: classes.loading }}
+            >{`Fetching Product Options...`}</LoadingIndicator>
+        );
+    }
+
+    return (
+        <Form
+            getApi={setFormApi}
+            initialValues={{ quantity: cartItem.quantity }}
+            onSubmit={handleSubmit}
+        >
+            <Options
+                classes={{ root: classes.optionRoot }}
+                onSelectionChange={handleOptionSelection}
+                options={configItem.configurable_options}
+                selectedValues={cartItem.configurable_options}
+            />
+            <h3 className={classes.quantityLabel}>Quantity</h3>
+            <QuantityFields
+                classes={{ root: classes.quantityRoot }}
+                initialValue={cartItem.quantity}
+                itemId={cartItem.id}
+            />
+            <div className={classes.submit}>
+                <Button priority="high" type="submit">
+                    Update
+                </Button>
+            </div>
+        </Form>
+    );
+};
+
+export default ProductForm;
+
+export const GET_CONFIGURABLE_OPTIONS = gql`
```

Comment:
Can we use [this query](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/queries/getProductsBySku.graphql) here instead?

Reply:
We're getting away from shared queries at the app root and want to colocate them to their components. We are okay with sharing fragments though, but we haven't quite worked out how those would be colocated and shared.